dbt/adapters/athena/impl.py

```diff
@@ -44,11 +43,28 @@ def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
     @available
-    def s3_uuid_table_location(self):
+    def generate_s3_data_path(
+        self,
+        env_name: str,
+        domain_name: str,
+        schema_name: str,
+        table_name: str,
+        run_time: Optional[str] = None,
+    ) -> str:
         conn = self.connections.get_thread_connection()
-        client = conn.handle
+        client = conn.credentials
 
-        return f"{client.s3_staging_dir}tables/{str(uuid4())}/"
+        if run_time is None:
+            return (
+                f"{client.s3_data_dir}/{env_name}/domain_name={domain_name}/"
```

Comment:
I didn't sleep and it will take me longer to check this than to ask: the previous version had `{s3_staging_dir}tables`, suggesting that the client attribute ended in a `/`. But for `s3_data_dir` we include the `/`, suggesting it isn't already in the attribute. Is this correct?

Reply:
Yep, that's right. The `s3_staging_dir` that already exists in the dbt profile is written that way, so I followed the same convention.
I'll need to come back to this function at some point because we could modify both the `s3_staging_dir` and `s3_data_dir` to just take the bucket name. I think that would be cleaner.
There's also the fact that the use of `run_time` in the path isn't behaving as expected. Instead of writing the data to a new location, and so preserving the previous deployment, it overwrites the existing and writes out with the new run time in the path. That won't affect user testing while I'm away, but it'll be something to get right when I'm back.